docs/source/ProgrammingGuide/Graph.rst

```diff
@@ -57,6 +57,52 @@ There are many advantages. Here are a few:
   the backend driver and/or compiler can leverage optimization opportunities.
 * Launch overhead is reduced, benefitting DAGs consisting of small workloads.
 
+Capture
+~~~~~~~
+
+It might be necessary to add nodes to a :cppkokkos:`Kokkos::Graph` whose workloads are not defined
+using any :cppkokkos:`Kokkos` code.
+This is the case *e.g.* when calling external math libraries like `cuBLAS`.
+
+Such a scenario can be encountered in many situations like building and training a neural network,
+running a conjugate gradient method, and so on.
+
+Capturing into a :cppkokkos:`Kokkos::Graph` boils down to writing the following snippet:
+
+.. code:: c++
+
+    auto my_captured_node = predecessor.cuda_capture(
+        exec,
+        [data = data](const Kokkos::Cuda& exec_) { ... }
+    );
+
+When the node is added to the :cppkokkos:`Kokkos::Graph`, the workloads are not dispatched to the device.
+Rather, the backend operations are "saved" for later "reuse" in the *capture* node.
+
+This is achieved by setting the underlying "stream" in "*capture* mode" before invoking the function object,
+such that backend operations are saved to the *capture* node instead of being enqueued in the passed execution space instance.
+
+Some important aspects of *capture* are worth pointing out:
+
+1. The function object will be stored by the :cppkokkos:`Kokkos::Graph` instance,
+   thereby ensuring that any data bound to the function object is guarenteed to
+   stay alive until the graph is destroyed.
+2. The parameter `exec` can be used to assign the captured workloads to the device associated with
+   `exec`.
+3. The *capture* mechanism will temporarily alter the `exec` state by setting the underlying "stream" in "*capture* mode", such that
```

Comment:
What are the documentation references for the other backends?

Reply:
* `SYCL` https://github.com/intel/llvm/blob/ee5e1ca95c78576c1b6f12b1c2d461ef4b537a9b/sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc?plain=1#L167-LL170 is a good starting point, and the link will be valid forever so I'm fine adding it.
* `HIP`: https://rocm.docs.amd.com/projects/HIP/en/latest/reference/hip_runtime_api/modules/graph_management.html#_CPPv421hipStreamBeginCapture11hipStream_t20hipStreamCaptureMode but the link seems pretty fragile.

Reply:
Hmm I prefer having the links and updating them if someone complains later.